Added persistence to LLDP changes

cmagnobarbosa opened the pull request:
Now, the LLDP status changes is stored in the NApp storehouse.

Main Improvements:

- Add an event indicating administrative changes to the LLDP.



Related:
- Issue Fix #49 
- PR in topology  https://github.com/kytos/topology/pull/111

cmagnobarbosa committed:
Added persistence to LLDP changes

Now, the LLDP status changes is stored in the NApp storehouse.

ajoaoff commented on main.py:
```diff
@@ -183,6 +183,11 @@ def notify_uplink_detected(self, event):
 
+    def notify_lldp_change(self):
+        """Dispatch a KytosEvents to notify changes to the LLDP status."""
+        event_out = KytosEvent(name='kytos/topology.network_status.updated')
```
This event is created by `kytos/lldp`, so its name should be `kytos/lldp....` instead of `kytos/topology....`.

cmagnobarbosa replied:
Thanks for the review, resolved in comment 608c17b.

cmagnobarbosa committed:
Fixed kytos Event Name.

hdiogenes commented on main.py:
```diff
@@ -183,6 +183,11 @@ def notify_uplink_detected(self, event):
             self.controller.buffers.app.put(event_out)
 
+    def notify_lldp_change(self):
+        """Dispatch a KytosEvents to notify changes to the LLDP status."""
```
```suggestion
        """Dispatch a KytosEvent to notify changes to the LLDP status."""
```

hdiogenes commented on main.py:
```diff
@@ -337,6 +342,7 @@ def disable_lldp(self):
             else:
                 error_list.append(id_)
         if not error_list:
+            self.notify_lldp_change()
```
It would be better if the event was not an empty one. It could include some information about which interface was modified, and if it was enabled or disabled.

cmagnobarbosa replied:
I added information about the modified interface at the event. These modifications also changed the PR https://github.com/kytos/topology/pull/111

cmagnobarbosa committed:
Add more information about the interfaces in the persistence Event

The Event responsible for notifying the persistence module now contains
information about the modified interfaces, allowing a log with more details.

hdiogenes commented on main.py:
```diff
@@ -324,6 +333,7 @@ def get_lldp_interfaces(self):
     def disable_lldp(self):
         """Disables an interface to receive LLDP packets."""
         interface_ids = self._get_data(request)
+        changed_interfaces = interface_ids
```
Move this line to the line just below the modification (`interface.lldp = False`).

hdiogenes commented on main.py:
```diff
@@ -349,6 +361,7 @@ def disable_lldp(self):
     def enable_lldp(self):
         """Enable an interface to receive LLDP packets."""
         interface_ids = self._get_data(request)
+        changed_interfaces = interface_ids
```
Same here

hdiogenes commented on main.py:
```diff
@@ -337,6 +347,8 @@ def disable_lldp(self):
                 error_list.append(id_)
         if not error_list:
+
+            self.notify_lldp_change('disabled', changed_interfaces)
```
This must be inside a new `if changed_interfaces`, not inside `if not error_list`

cmagnobarbosa committed:
Now, the persistence event sends only changed interfaces
